qcodes/instrument/parameter.py

```diff
@@ -94,83 +94,89 @@ class _BaseParameter(Metadatable, DeferredOperations):
             ``update=True``, for example if it takes too long to update.
             Default True.
 
+        max_val_age (Optional[int]): The max time (in seconds) to trust a
+            saved value obtained from get_latest(). If this parameter has not
+            been set or measured more recently than this, perform an
+            additional measurement.
+
         metadata (Optional[dict]): extra information to include with the
             JSON snapshot of the parameter
     """
     def __init__(self, name, instrument, snapshot_get, metadata,
-                 snapshot_value=True):
+                 step=None, scale=None, inter_delay=0, post_delay=0,
+                 val_mapping=None, get_parser=None, set_parser=None,
+                 snapshot_value=True, max_val_age=None, vals=None):
         super().__init__(metadata)
-        self._snapshot_get = snapshot_get
         self.name = str(name)
         self._instrument = instrument
+        self._snapshot_get = snapshot_get
         self._snapshot_value = snapshot_value
 
-        self.has_get = hasattr(self, 'get')
```

Comment:
Aside from unit tests which have to be modified, I found one other instance in QCoDeS that uses has_get and has_set. I changed that in the latest commit. 

I can't say if other forks of QCoDeS use has_get/set. To me it looks like an attribute that is not used much, but I could keep it as a deprecated attribute, I'm fine either way. What do you suggest?